torch/jit/__init__.py

```diff
@@ -280,7 +269,10 @@ def get_trace_graph(f, args=(), kwargs=None, _force_outplace=False,
         kwargs = {}
     if not isinstance(args, tuple):
         args = (args,)
-    return ONNXTracedModule(f, _force_outplace, return_inputs, _return_inputs_states)(*args, **kwargs)
+    torch.jit._trace_module_map = {}
```

Comment:
You get _one_ place where this global is set/reset.  It is really hard to know where it has to be manipulated at the moment because it exists in three different places. Pass it in as an argument up until it is no longer feasibly to do so.